pkg/common/k8s_utils.go

```diff
@@ -35,10 +35,15 @@ const (
 	ReadyStatusConditionType = "Ready"
 )
 
+// ObjectInfo generates a string representation of the provided Kubernetes object, including its kind and name.
+// The object must implement the client.Object interface, which provides access to the object's metadata.
```

Comment:
nitpick: this is a personal taste. Comments are usually not in excess. The more comments, the better. But I tend to add comments only for "not obvious" things.